joomla/joomla-cms
# [language filter] Add HTTP Redirect code parameter (default to 302 Found)

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/11165.
#### Summary of Changes

This PR adds a new option to the language filter plugin where the user can select the redirect HTTP status code. The default value for this option is a "302 Found" redirect.

This means, that by default all language filter redirects will be a 302 redirect, users can now change the redirect code in the language filter plugin options.
#### Testing Instructions
1. Use latest staging
2. Apply patch
3. Check the default language filter redirect code is a 302 in all cases
4. Change it to a 301 redirect in the language filter plugins options and check the redirect code is a 301 redirect code (not cachable) in all cases.
5. Do a code review
#### More info

This will allow greater flexibility for those who want to change the redirect code, a similiar option exists in com_redirect.

See https://tools.ietf.org/html/rfc7231#section-6.4 (300 to 307 redirects) and https://tools.ietf.org/html/rfc7238 (308 redirect)

@ggppdk @infograf768 @Bakual
